### ISSUE TITLE
Docs on Reactivity/Events from selections in charts

### ISSUE THREAD

**Issue by joelostblom** [posit-dev/py-shiny#781]
Does shiny support reactivity/events from making a selection in an interactive chart such as Altair, Plotly, Bokeh, etc? I can't seem to find any examples of this; the closest is the page on "Reacting to widget updates" seem to suggest that shiny only supports events from selections in charts that provide this via an already existing jupyter widgets interaction (such as `on_click` etc), is this correct?

In Altair, we recently added support for exactly this: [chart selections can be accessed in notebooks through a jupyter widget chart object](https://altair-viz.github.io/user_guide/jupyter_chart.html#selection-params). Based on my understanding this would be supported in shiny too (and maybe even quarto dashboards since they also support jupyter widgets?), but I want to double check if this is the case before trying it out.

I think it could be valuable to add a section on chart selections driving events in the docs, just to be explicit about what works and what doesn't.

**Comment by cpsievert** [comment]
> Does shiny support reactivity/events from making a selection in an interactive chart such as Altair, Plotly, Bokeh, etc?

Yes, `{shinywidgets}`'s `reactive_read()` function supports reactively reading trait(s) of any ipywidget that supports an [`.observe()` method](https://ipywidgets.readthedocs.io/en/stable/examples/Widget%20Events.html#traitlet-events). In fact, in that ["Reacting to widget updates" ](https://shiny.posit.co/py/docs/ipywidgets.html#reactive-read) section you reference, the `reactive_read(map, "center")` line essentially does a reactive version of `map.observe(callback, "center")`

[Here is how you can leverage `reactive_read()` to respond to altair selections](https://shinylive.io/py/editor/#code=NobwRAdghgtgpmAXGKAHVA6VBPMAaMAYwHsIAXOcpMAMwCdiYACAZwAsBLCbJjmVYnTJMAgujxMArhwl1KAEzh1ZcKITIcAbnAA6Eeo1aduAdw7yA5nDIte-QcLkWOLCnQD6Zy9YnFJZVH9PcysyFTUNbXc5KHk9PgEhJigAGzIoDjpk21SyPQNmbQsod3kodJZrWwSHJjL0vT0WPzpCOCYAXjryqAxCKDoWAAoASkbodHdpTqkOLCgrdxoU6XkhvSZN2Yw-AKCKAA8yd206ACNyvnWwSpS4dQ5SHTARvA2t3cDjr1DrwjYBnkXnoxhA9IoaKwlKchlwvr5-PCoSwWI8ICNEOMtkwAMRMADCMQoTDIbHaAClJDg3PiAUkCkx3ps8WwyAEWIgAPSc3IZOgAWk0HAAXhhnKTJGcMI9OZJKh4LKs4JyAFZU7Budz-QEYVkwFI4gRcMj8273DSkFhMphnOhytgzXIYM0PUjuI3kIbQeAdZ4eoESSgkeRcCwsDrAZ4kFKCZ4AXQkZy48l9YDuVggcWBEGx2qSXSdtMBQ2aklacBGGBgAwA1u7iMbRhgg8RFOsc9imAcOgByAASgkqAhMSkQAEUe28O9jsL2ALIcO4sd1KdwAcVSMYg48n1s20cEHSdJEzHAtECGtvtEh7AHk6BxnNuAHKT5JpDCaVKSOBDHsWORsB7EZXmtStYnkd0BlgYYr3YUFsRVPNhALD9KWpJQiyEIZkNBa08QAERcVAUigbBOScFw3BJMkmGQ3gc1Jdo0FQKYOEZacmEo1xVx+aw-jpAMmCQwS8M4vEABUHwsKwshiB4hTIHgTDJCA4FOZIoTuV0c2IQhCDLK1OIAAU+fxrWMuRMyUDBDjyTiIS0800VGTFOOxVAUK41QFLgaJVDWESdRdc8WAkP0G3IZ4EM7TZUAfT1PJi2K5DIMsc2eABlOBtIoeQmH9RBGTAJgAGpWDIOghiS8YWJmMRUCGFi2IkeVThGMAAF88HAb0EGQOQAEdpDkeByBYWyjnwIhSAoKhkCgUwQmsPReUyPQihKeooEqGwurjIA), in a similar fashion in [the link you provided](https://altair-viz.github.io/user_guide/jupyter_chart.html#selection-params) (notice how clicking on the altair legend reactively triggers code to run).

It was slightly tricky to get this working since I was expecting `reactive_read(jchart, "selections")` to work, but it doesn't since `jchart.observe(callback, "selections")` doesn't seem to fire `callback` when the selection changes? Kind of seems like that's oversight/issue on altair's part?

**Comment by joelostblom** [comment]
This is super, thank you for the detailed example @cpsievert ! I will start experimenting with creating some Shiny + Altair dashboards on my own and see if I run into any limitations, but it looks pretty convenient from your example. I think it would be helpful to add that example to the docs and I also raised the issue you brought up in the Altair repo to see if there is something we can modify there to make the Shiny syntax more convenient.
